crates/astria-conductor/src/celestia/reconstruct.rs

```diff
@@ -0,0 +1,201 @@
+use std::collections::HashMap;
+
+use astria_core::sequencer::v1::RollupId;
+use celestia_client::{
+    CelestiaRollupBlob,
+    CelestiaSequencerBlob,
+};
+use sequencer_client::tendermint::block::Height as SequencerHeight;
+use telemetry::display::base64;
+use tracing::{
+    info,
+    warn,
+};
+
+use super::{
+    verify::VerifiedBlobs,
+    ReconstructedBlock,
+};
+
+/// Reconstructs block information from verified blocks.
+///
+/// The reconstructed blocks contain a block hash, tendermint header (both extracted from
+/// sequencer header blobs), and transactions (from rollup blobs, if present).
+///
+/// This process works in the following way:
+/// 1. Header blobs are placed in a map using a key `(block_hash, height)`.
+///    - header blobs with the same `(block_hash, height)` are dropped.
```

Comment:
height is sequencer height, not celestia height right? as per the comment on `reverse_index`

Reply:
removed any mention of heights from this reconstruction logic (no longer relevant because it was absorbed by `verify`).